### PR TITLE
serial-unit-testing: init at 0.2.4

### PR DESCRIPTION
**Pull request by rudolfvesely**
Package "serial-unit-testing" is a cross-platform cli application and rust library. Using serial unit testing communication and testing with any serial port device can be done and most important automated. Release notes: https://github.com/markatk/serial-unit-testing/blob/v0.2.4/CHANGELOG.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested, as applicable
  - Compiled and run against a serial port - fully functional.
  - Unit tests require an accessible serial port and therefore skipped by: `doCheck = false;`
- [x] Tested compilation of all packages that depend on this change:
  - No dependencies except what is in `nativeBuildInputs` and `buildInputs`
- [x] Tested basic functionality of all binary files: Yes

### CODE REVIEW

File: pkgs/by-name/se/serial-unit-testing/package.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  # tests require a serial port
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Automate testing of serial communication with any serial port device.";
+    homepage = "https://github.com/markatk/serial-unit-testing";
+    changelog = "https://github.com/markatk/serial-unit-testing/blob/v${version}/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = with maintainers; [ rudolfvesely ];
```
Comment (nagy) [inline pkgs/by-name/se/serial-unit-testing/package.nix:37]: As you can see by the failing ofborg evaluation, this maintainer attribute would need to be added into the `maintainer-list.nix` file ( in a separate commit ).

**Review by nagy — CHANGES_REQUESTED** [review]
Can you also follow the commit message guidelines at https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions ( and name the PR title accordingly as well) .

File: pkgs/by-name/se/serial-unit-testing/package.nix (hunk 1 above)
Comment (rudolfvesely) [reply pkgs/by-name/se/serial-unit-testing/package.nix:37]: Thank you for the review. I changed the message and added myself as a maintainer.

**Review by nagy — CHANGES_REQUESTED** [review]
The commit message of the package is still not following the guidelines. It should be the same as the PR title is now. Also we dont need the merge commit, you can squash/drop it.

**Comment by rudolfvesely** [comment]
Thank you. That's done now.

File: pkgs/by-name/se/serial-unit-testing/package.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "serial-unit-testing";
+  version = "0.2.4";
+
+  src = fetchFromGitHub {
+    owner = "markatk";
+    repo = "serial-unit-testing";
+    rev = version;
```
Comment (nagy) [inline pkgs/by-name/se/serial-unit-testing/package.nix:15]: I get 
```
error: builder for '/nix/store/c7bwdlzig3kq0m1qlgxvxdivqqig2rgp-source.drv' failed with exit code 1;
       last 8 log lines:
       >
       > trying https://github.com/markatk/serial-unit-testing/archive/0.2.4.tar.gz
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       >   0    14    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 404
       > error: cannot download source from any mirror
       For full logs, run 'nix log /nix/store/c7bwdlzig3kq0m1qlgxvxdivqqig2rgp-source.drv'.
error: 1 dependencies of derivation '/nix/store/j8ysbkvc9g6gji5bmp8a5550zdzq8rjw-serial-unit-testing-0.2.4.drv' failed to build
```

You probably need to use `rev = "v${version}";`

**Review by nagy — CHANGES_REQUESTED** [review]

File: pkgs/by-name/se/serial-unit-testing/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  # tests require a serial port
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Automate testing of serial communication with any serial port device.";
+    homepage = "https://github.com/markatk/serial-unit-testing";
+    changelog = "https://github.com/markatk/serial-unit-testing/blob/v${version}/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = with maintainers; [ rudolfvesely ];
```
Comment (nagy) [inline pkgs/by-name/se/serial-unit-testing/package.nix:37]: It builds now. Can you add `mainProgram = "sut";` here? after that , its good.

**Review by nagy — CHANGES_REQUESTED** [review]

File: pkgs/by-name/se/serial-unit-testing/package.nix (hunk 3 above)
Comment (rudolfvesely) [reply pkgs/by-name/se/serial-unit-testing/package.nix:37]: Done. Thank you for the review.

**Review by nagy — APPROVED** [review]

File: pkgs/by-name/se/serial-unit-testing/package.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  buildInputs = [
+    udev
+  ];
+
+  # tests require a serial port
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Automate testing of serial communication with any serial port device.";
```
Comment (sikmir) [inline pkgs/by-name/se/serial-unit-testing/package.nix:33]:
```suggestion
    description = "Automate testing of serial communication with any serial port device";
```
See https://nixos.org/manual/nixpkgs/stable/#var-meta-description

**Review by wegank — APPROVED** [review]

**Pull request merged**